src/app.ts

```diff
@@ -1,14 +1,22 @@
 import express, { Express } from 'express';
 
+import UserService from './services/users.service';
+import usersRouter from './routes/users.router';
+import UsersMiddleware from './middleware/users.middleware';
+import UsersController from './controllers/users.controller';
+import UserRepository from './repositories/userRepository';
 import type { DataBaseI } from './types/database';
 
-// TODO: pass DB to repository
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 function app (db: DataBaseI): Express {
-
   const application: Express = express();
 
+  const userRepository: UserRepository = new UserRepository(db);
```

Comment:
Consider refactoring to dependency injection for better testability and maintainability. This will allow you to easily swap out dependencies and reduce code coupling. You can use a library like [awilix](https://www.npmjs.com/package/awilix) for this purpose.